apps/wallet-mobile/src/features/Swap/common/helpers.ts

```diff
@@ -66,3 +70,38 @@ export const useCancelOrderWithHw = (
     cancelOrder: mutation.mutate,
   }
 }
+
+export type OrderTxMetadata = {
+  sellTokenId: string
+  buyTokenId: string
+  sellQuantity: string
+  buyQuantity: string
+  provider: string
+}
+
+const OrderTxMetadataSchema: z.ZodSchema<OrderTxMetadata> = z.object({
+  sellTokenId: z.string(),
+  buyTokenId: z.string(),
+  sellQuantity: z.string(),
+  buyQuantity: z.string(),
+  provider: z.string(),
+})
+
+const isOrderTxMetadata = createTypeGuardFromSchema(OrderTxMetadataSchema)
+
+/**
+ * Parses and validates a JSON metadata string, transforming it into a structure compliant with MappedRawOrder['metadata'].
+ *
+ * @param metadataJson - The JSON string representation of metadata.
+ * @returns The parsed metadata object or null if parsing fails or validation fails.
+ */
+export const parseOrderTxMetadata = (metadataJson: string): OrderTxMetadata | null => {
+  try {
+    const metadata = JSON.parse(metadataJson)
+
+    return isOrderTxMetadata(metadata) ? metadata : null
+  } catch (error) {
+    Logger.warn('JSON parsing error:', error)
+    return null
+  }
```

Comment:
we have the `parseSafe` from `@yoroi/common` 
```suggestion
  const parsed = parseSafe(metadataJson)
  return isOrderTxMetadata(metadata) ? metadata : null
  }
```